_posts/2021-10-12-devfile-v2-and-the-devworkspace-operator-p1.md

```diff
@@ -0,0 +1,149 @@
+```yaml
+schemaVersion: 2.1.0
+metadata:
+  name: python-hello-world
+attributes:
+  che-theia.eclipse.org/sidecar-policy: USE_DEV_CONTAINER
+components:
+  - name: python
+    container:
+      image: quay.io/devfile/base-developer-image:ubi8-7bd4fe3
+      volumeMounts:
+        - name: venv
+          path: /home/user/.venv
+      memoryLimit: 512Mi
+      mountSources: true
+  - name: venv
+    volume:
+      size: 1G
+```
+
+The Devfile specification has gone through the release of v2. Here are a few notable changes:
+- It is compatible with the specification of a Kubernetes API. The DevWorkspace CRD is an extension of the Kubernetes API and it’s generated from the Devfile specification.
+- It removes [chePlugin and cheEditor component types](https://github.com/eclipse/che/issues/18669).
+- It introduces the volume component type.
+- Events and parent are two new components types.
+- Besides Che it’s used by the OpenShift Developer Console, `odo` and the [Devfile Docker plugin](https://github.com/devfile/devfile-docker-plugin). 
+
+The documentation for Devfile v2.1 is [https://devfile.io/docs/devfile/2.1.0/](https://devfile.io/docs/devfile/2.1.0/user-guide/index.html) and a migration to v2 guide 
+
+### A new way to specify the editor and its plugins
+As mentioned above, version 2 of the Devfile, doesn’t include cheEditor and chePlugins component types anymore. 
+
+The recommended way to specify the editor of a workspace is to include the file `.che/che-editor.yaml` at the root of the workspace git repository:
+```yaml
+id: eclipse/che-theia                 # mandatory
+registryUrl: https://my-registry.com  # optional
+override:                             # optional
+  containers:
+    - name: theia-ide
+      memoryLimit: 1280Mi
+```
+
+The recommended way to specify a che-theia plugin in a workspace is to include the file `.vscode/extensions.json` at the root of the workspace git repository:
+```json
+{
+    "recommendations": [
+      "ms-python.python"
+    ]
+}
+```
+ 
+It is also possible to define a Che editor and its plugins inline in a Devfile attributes. But that’s not recommended. More informations can be found at [https://github.com/eclipse/che/issues/18669](https://github.com/eclipse/che/issues/18669).
+
+### The Devfile should live at the root of the git repo, not in a registry
+The recommended place to publish the Devfile is within the project source code. Along with the files that we have just seen above to specify che-theia plugins and the editor:
+
+```
+  |--- devfile.yaml
+  |___ .che
+         |___ che-editor.yaml
+         |___ che-theia-plugins.yaml
+  |___ .vsocode
+         |___ extensions.json
+```
+
+The Devfile should be versioned and its version should be synchronized with the version of the source code. Having the Devfile at the root of a repository makes it possible to use a simple factory link to the repository to start the workspace.
+
+Versioning the Devfile with the source code has two consequences: 
```

Comment:
Ok, I think my confusion comes from a different understanding of "versioning". I would understand this more clearly if instead of versioning this sentence used "Colocating and evolving the devfile together with the source code ...".